### DISCUSSION TITLE
remove red "Development" tag

### DISCUSSION THREAD

**Discussion by yskimCal** [apache/superset#27563] [Q&A / Help] [answered]
Hello,

Does anyone know how to remove red "Development" tag in UI..?

<img width="1783" alt="Screenshot 2024-03-19 at 5 42 20 PM" src="https://github.com/apache/superset/assets/59595561/7aa64481-be18-4d60-8c8b-587e9e0080a0">

Thank you!

**Comment by sfirke** [comment, marked as answer]
I wonder if you need to add one of these values in your config -- please try one at a time and let us know which worked:
```
FLASK_ENV=production
SUPERSET_ENV=production
```

**Reply by yskimCal** [reply]
Thank you for your response!

However, unfortunately, both did not worked.. It said 

> NameError: name 'production' is not defined

**Comment by inkerinmaa** [comment]
I set SUPERSET_ENV=production in .ENV file and SUPERSET_TESTENV=false in docker-init.sh. It works

**Reply by persistanceTube** [reply]
it worked for me  
version 4.0.2

**Reply by MohammadRezaAlikhani** [reply]
> I set SUPERSET_ENV=production in .ENV file and SUPERSET_TESTENV=false in docker-init.sh. It works

Thanks a lot, I deployed version 4.1.2 and it worked for me too. I don't know if some problems will happen in the future but, it's working fine.

**Comment by Unanimad** [comment]
Hello everyone. None of these worked for me :/ Version 3.1.2 and 3.1.1

Edit: Set variables in `./docker/.env`

**Comment by mistercrunch** [comment]
I also tested confirmed that adding a line `SUPERSET_ENV=production` in either `docker/.env-local` or `docker/.env` removes the tag. We discourage from using docker-compose in production though, those files are the for development and sandbox envrionment.

**Reply by Neang-Rothmny** [reply]
Hello,
How do we deploy to production correctly?
Thank you!!

**Reply by inkerinmaa** [reply]
And what's wrong with docker compose in production? Why should we avoid it?

**Reply by sfirke** [reply]
@inkerinmaa there's some info here: https://superset.apache.org/docs/installation/docker-compose/. Some of the reasons to avoid it may not apply to you (for instance, Kubernetes can scale up and down and heal containers, docker compose can't but maybe you don't need that). Others are universal precautions, for instance it's risky to have your application metadata stored in a Postgres DB in a Docker container rather than a permanent, backed up and managed DB. And it's potentially risky to run your docker application as root.

That said, if you think you understand the risks and are comfortable with them and can mitigate them, it's certainly an option.

**Reply by mistercrunch** [reply]
I'll add one more thing, from a maintainer's perspective, having clear lines drawn between "this is meant for development" and "this other thing is meant for production" really helps staying sane and for common mistakes to be made. For instance in development is desirable to have `root` access in the image, and to expose a not-so-lean container with a bunch of dev tools available in the container.

Reusing the same foundation and having a bunch of flags/env-vars/alternate-configs/alternate-layers makes things really complicated (too many switches, buttons and levers make things unmanageable). Having more segmentation allows us to optimize for each use case, without crumbling under the weight of configurability.

Now knowing all this, you decide what's best given your the constraints and goals you have.

For lean production set up, I'd recommend giving minikube a shot, you get the bulk of the value you'll get from using a proper orchestrator while staying in a results-first/move-fast mindset.

**Reply by mistercrunch** [reply]
I'll also say that if your organization doesn't offer a modern-infra-framework (k8s, observability tooling, ...) environment to productionize internal tools, you may want to consider a hosted service [shameless plug for www.preset.io] here, as we serve Superset for orgs of all sizes, and most likely can run Superset cheaper and more reliably than you can on your own.
